test/test_fcl_coplanar_triangles.cpp

```diff
@@ -0,0 +1,243 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2017, Georgia Tech Research Corporation
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Georgia Tech Research Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/** \author Andrew Price <arprice@gatech.edu> */
+
+#include <gtest/gtest.h>
+
+#include "fcl/fcl.h"
+#include "fcl/math/constants.h"
+#include "test_fcl_utility.h"
+#include "test_fcl_hungarian.h"
+#include "test_fcl_hungarian-inl.h"
+
+using namespace fcl;
+
+template <typename Scalar>
+void verifyContacts(const Vector3<Scalar>* expected, const Vector3<Scalar>* actual, const int numContacts)
+{
+  typedef Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic> CostMatrix;
+
+  CostMatrix C(numContacts, numContacts);
+  for (int i = 0; i < numContacts; ++i)
+  {
+    for (int j = 0; j < numContacts; ++j)
+    {
+      C(i,j) = (expected[i]-actual[j]).squaredNorm();
+    }
+  }
+
+  fcl::test::Hungarian<Scalar> H(C);
+
+  std::vector<int> permutation = H.getAssignment();
```

Comment:
Could you achieve the same effect here by sorting `expected` and `actual` with `std::sort` and a custom comparison? Adding the Hungarian algorithm just to get the corresponding points seems like overkill. The following comparison should work, though `expected` and `actual` would need to be modified to be `std::vector<Vector3<Scalar>>`. Of cource, it's also quite possible that I'm missing something.
```c++
template <typename Scalar>
bool comparePoints(const Vector3<Scalar>& v1, const Vector3<Scalar>& v2)
{
  for (int i = 0; i < 3; ++i) {
    if (v1[i] < v2[i]) {
      return true;
    } else if (v1[i] > v2[i]) {
      return false;
    }
  }
  // In this case v1 == v2
  return false;
}
```

Reply:
Yeah, this is probably overkill. I had that code lying around from something else, so I figured it was easiest to just do the matching the "right" way, but only have the class in the test methods so as to not expose it to the main body of code. Is it a problem to have it there?